src/kmeans.jl

```diff
@@ -1,91 +1,132 @@
 # K-means algorithm
 
-#### Interface
-
-mutable struct KmeansResult{T<:Real} <: ClusteringResult
-    centers::Matrix{T}         # cluster centers (d x k)
-    assignments::Vector{Int}   # assignments (n)
-    costs::Vector{T}           # costs of the resultant assignments (n)
-    counts::Vector{Int}        # number of samples assigned to each cluster (k)
-    cweights::Vector{Float64}  # cluster weights (k)
-    totalcost::Float64         # total cost (i.e. objective) (k)
-    iterations::Int            # number of elapsed iterations
-    converged::Bool            # whether the procedure converged
+####
+#### Result object
+####
+
+# NOTE REFACT: there's no need for this to be a mutable struct.
+
+struct KmeansResult{T<:AbstractFloat} <: ClusteringResult
```

Comment:
Actually, rather then tightening the constraint, I think it could be relaxed even further to `<: Number` (i.e. complex numbers).
Anyway, the inference and/or checking of `T` would be done by `kmeans[!]()`.

Reply:
Do you have a reference/use-case for kmeans on Complex Number? This could be for another PR though...
Here, I tightened the parameter of that type because it never got anything else.